docs/how-tos/caching.rst

```diff
@@ -0,0 +1,90 @@
+========
+Caching
+========
+
+In Hamilton, the term *caching* broadly refers to "reusing results from previous executions to skip redundant computation". When caching is enabled, node results are stored in a **result store** and where they can be retrieved by subsequent runs.
+
+.. seealso::
+
+    For a broader overview of how caching works, see the :doc:`/concepts/caching` Concepts page .
+
+    For implementation details, see the :doc:`/reference/caching/caching-logic` Reference page .
+
+
+Use cases
+-----------
+
+Caching is beneficial for almost all use cases, and it's likely a good idea to have it on. The next sections present a illustrative scenarios.
+
+
+Iterative notebook development
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You are developing a dataflow in a notebook (analytics, ETL, ML pipeline, LLM workflow, etc.). You want to explore and iterate quickly over different transformation steps. For example:
+
+.. code-block:: python
+
+    import pandas as pd
+    from sklearn.linear import LinearRegression
+
+    def raw_data(file_path: str) -> pd.DataFrame:
+        return pd.read_parquet(file_path)
+
+    def cleaned_data(raw_data: pd.DataFrame) -> pd.DataFrame:
+        cleaned_df = ...  # fill nulls, filter rows, rename columns
+        return cleaned_df
+
+    def statistics(cleaned_data: pd.DataFrame) -> dict:
```

Comment:
Add comments to say what users will change. E.G. you want to keep the same raw data but mess around with cleaning...

Reply:
But yeah, these don't capture all the use-cases we discussed/the more horizontal one (iterating on a node, changing the data, changing the parameters, etc...) I think just "getting started" is a good way, unless you're already planning to rewrite this.

Reply:
Now covered in the tutorial notebook